tuning/trainercontroller/controllermetrics/eval_metrics.py

```diff
@@ -0,0 +1,65 @@
+# Copyright The IBM Tuning Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# SPDX-License-Identifier: Apache-2.0
+# https://spdx.dev/learn/handling-license-info/
+
+# Standard
+from typing import Any
+
+# Third Party
+from transformers.utils import logging
+
+# Local
+from tuning.trainercontroller.controllermetrics.metricshandler import MetricHandler
+
+logger = logging.get_logger(__name__)
+
+
+class EvalMetrics(MetricHandler):
+    """Implements the controller metric which exposes the evaluation metrics"""
+
+    def __init__(self, **kwargs):
+        """Initializes the metric handler, by registering the event \
+            list and arguments with base handler.
+
+        Args:
+            kwargs: List of arguments (key, value)-pairs
+        """
+        super().__init__(
+            events=[
+                kwargs.get("source-event", "on_evaluate"),
```

Comment:
Could you explain the need for the source-event arg here, instead of just hardcoding the event to `on_evaluate`? I.e., aren't the metrics in the trainer callback only available for `on_evaluate` ([ref](https://huggingface.co/docs/transformers/en/main_classes/callback#transformers.TrainerCallback.metrics))?

Reply:
@alex-jw-brooks We put this line to cater to `on_predict` event as well. It seems metrics are passed to this event too (please refer to [link](https://github.com/huggingface/transformers/blob/4fdf58afb72b0754da30037fc800b6044e7d9c99/src/transformers/trainer_callback.py#L283)).

Reply:
Ahh that is really interesting, maybe that section of the documentation saying it is only accessible from eval is outdated, and I guess custom callbacks can kind of do what they want anyway, so the metrics provided to those events might not always be the same 🤔 Thank you for the explanation, makes sense